src/libraries/Microsoft.Extensions.Options/tests/Microsoft.Extensions.Options.Tests/OptionsTest.cs

```diff
@@ -167,6 +167,18 @@ public void PostConfiguresInRegistrationOrderAfterConfigures()
             Assert.Equal("_-ABC", option.Message);
         }
 
+        [Fact]
+        public void SingleValidateOnStartRegistrationTest()
```

Comment:
Can we also add a test that validates that we can add some other implementation of `IStartupValidator` to the services collection?  For example - some user implements their own `IStartupValidator` and some library is also using `ValidateOnStart`.

Reply:
Oh, nevermind -- looks like there can only be one: https://github.com/tarekgh/runtime/blob/496f96efe52b1b446ed7b4b6ce9f269c85c0bf36/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L96

Reply:
Yeah, I tried this scenario and it's the last one registered that gets used.  No exceptions and no multiple service implementations allowed.